crates/analyzer-host/src/tests/unit_tests.rs

```diff
@@ -0,0 +1,62 @@
+mod fsm_tests {
+	use std::sync::Arc;
+
+	use serde_json::Value;
+
+	use crate::{
+		fs::LspEnumerableFileSystem,
+		fsm::LspProtocolMachine,
+		json_rpc::message::{Message, Notification, Request},
+		lsp::{request::RequestManager, state::LspServerState},
+	};
+
+	#[test]
+	fn test_lsp_states() {
+		let rm = RequestManager::new(async_channel::unbounded::<Message>());
+		let mut lsp =
+			LspProtocolMachine::new(None, rm.clone(), Arc::new(Box::new(LspEnumerableFileSystem::new(rm.clone()))));
+		assert_eq!(lsp.current_state(), LspServerState::ActiveUninitialized);
+		assert_eq!(lsp.is_active(), true);
+
+		let mut params = serde_json::json!(analyzer_abstractions::lsp_types::InitializeParams { ..Default::default() });
+		let mut message =
+			Message::Request(Request { id: 0.into(), method: String::from("initialize"), params: params });
+		let mut output = async_io::block_on(lsp.process_message(Arc::new(message)));
+		assert!(output.is_ok());
+		assert_eq!(lsp.current_state(), LspServerState::Initializing);
+		assert_eq!(lsp.is_active(), true);
+
+		params = serde_json::json!(analyzer_abstractions::lsp_types::InitializeParams { ..Default::default() });
+		message = Message::Request(Request { id: 0.into(), method: String::from("initialize"), params: params });
```

Comment:
If you wanted to test how the LSP will react in it's current state, you could pass an valid request but at an invalid state, instead of the same message, a request like: https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocument_hover. (could you add a default version to `tester` cargo)